cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/reserve_transfer.rs

```diff
@@ -1132,8 +1145,666 @@ fn reserve_transfer_native_asset_from_para_to_para_through_relay() {
+	)
+		.into()]
+	.into();
+	// Just to be very specific you don't need any native asset.
+	assert_eq!(assets.len(), 1);
+
+	// fund the Parachain Origin's SA on AssetHub with the native tokens held in reserve
+	AssetHubWestend::fund_accounts(vec![(
+		sov_of_sender_on_asset_hub.into(),
+		asset_amount_to_send * 2,
+	)]);
+
+	// Give the sender enough Relay tokens to pay for local delivery fees.
+	// TODO: When we support local delivery fee payment in other assets, we don't need this.
+	PenpalA::execute_with(|| {
+		assert_ok!(<PenpalA as PenpalAPallet>::ForeignAssets::mint(
+			<PenpalA as Chain>::RuntimeOrigin::signed(PenpalAssetOwner::get()),
+			Location::new(1, []),
+			sender.clone().into(),
+			10_000_000_000_000, // Big estimate to make it work.
+		));
+	});
+
+	// Init values for Parachain Destination
+	let receiver = PenpalBReceiver::get();
+
+	// Init Test
+	let fee_asset_index = 0;
+	let test_args = TestContext {
+		sender: sender.clone(),
+		receiver: receiver.clone(),
+		args: TestArgs::new_para(
+			destination,
+			receiver.clone(),
+			asset_amount_to_send,
+			assets,
+			None,
+			fee_asset_index,
+		),
+	};
+	let mut test = ParaToParaThroughAHTest::new(test_args);
+
+	// Query initial balances
+	let sender_assets_before = PenpalA::execute_with(|| {
+		type ForeignAssets = <PenpalA as PenpalAPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(system_para_foreign_asset_location.clone(), &sender)
+	});
+	let receiver_assets_before = PenpalB::execute_with(|| {
+		type ForeignAssets = <PenpalB as PenpalBPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(
+			system_para_foreign_asset_location.clone(),
+			&receiver,
+		)
+	});
+
+	// Set assertions and dispatchables
+	// TODO: Bring back assertions.
+	// test.set_assertion::<PenpalA>(para_to_para_through_hop_sender_assertions);
+	// test.set_assertion::<AssetHubWestend>(para_to_para_relay_hop_assertions);
+	// test.set_assertion::<PenpalB>(para_to_para_through_hop_receiver_assertions);
+	test.set_dispatchable::<PenpalA>(
+		para_to_para_through_asset_hub_limited_reserve_transfer_assets,
+	);
+	test.assert();
+
+	// Query final balances
+	let sender_assets_after = PenpalA::execute_with(|| {
+		type ForeignAssets = <PenpalA as PenpalAPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(system_para_foreign_asset_location.clone(), &sender)
+	});
+	let receiver_assets_after = PenpalB::execute_with(|| {
+		type ForeignAssets = <PenpalB as PenpalBPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(system_para_foreign_asset_location, &receiver)
+	});
+
+	// Sender's balance is reduced by amount sent plus delivery fees
+	assert!(sender_assets_after < sender_assets_before - asset_amount_to_send);
+	// Receiver's balance is increased
+	assert!(receiver_assets_after > receiver_assets_before);
+}
+
+// ===============================================================
+// ===== Reserve Transfers - Pool Asset - AssetHub->Parachain ====
```

Comment:
Also nice, but in practice, this random pool asset used to pay for fees on Parachains is either DOT or USDT, so I would use DOT/WND to cover this scenario.